libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1665,6 +1666,19 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         developerMetadataAggregator.unregisterObserver(developerMetadataObserver)
     }
 
+    /**
+     * Provide a [EVDataUpdater] that will notify Nav SDK of EV data updates
+     * to be used in route refresh requests.
+     * Set to null to stop providing updates.
+     * NOTE: after setting the updater to null the last saved state will be used in refresh requests.
+     *
+     * @param updater [EVDataUpdater] that will notify Nav SDK of EV data updates
+     */
+    @ExperimentalPreviewMapboxNavigationAPI
+    fun setEVDataUpdater(updater: EVDataUpdater?) {
```

Comment:
```suggestion
    fun onEVDataUpdated(data: Map<String, String?>) {
```
Why not simplify like this? What's the benefit of forcing the developers to implement the full observer pattern?

Reply:
I guess the initial purpose was to isolate the logic from `MapboxNavigation`. In case it became more complicated. So that I don't have to add 5 ev related methods to `MapboxNavigation`.
But it does require more actions from the user and it's only 1 method so I guess it makes sense to simplify.